Xamarin.PropertyEditing.Windows/GroupEditorControl.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Specialized;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Input;
+using System.Windows.Media;
+
+namespace Xamarin.PropertyEditing.Windows
+{
+	[TemplatePart (Name = "List", Type = typeof(ItemsControl))]
+	[TemplatePart (Name = "EditorPresenter", Type = typeof(Selector))]
+	internal class GroupEditorControl
+		: Selector
+	{
+		public GroupEditorControl ()
+		{
+			FocusableProperty.OverrideMetadata (typeof(GroupEditorControl), new FrameworkPropertyMetadata (false));
+
+			ItemContainerGenerator.StatusChanged += OnItemContainerGeneratorOnStatusChanged;
+		}
+
+		public static readonly DependencyProperty ContentTemplateProperty = DependencyProperty.Register (
+			"ContentTemplate", typeof(DataTemplate), typeof(GroupEditorControl), new PropertyMetadata (default(DataTemplate)));
+
+		public DataTemplate ContentTemplate
+		{
+			get { return (DataTemplate) GetValue (ContentTemplateProperty); }
+			set { SetValue (ContentTemplateProperty, value); }
+		}
+
+		public static readonly DependencyProperty ContentTemplateSelectorProperty = DependencyProperty.Register (
+			"ContentTemplateSelector", typeof(DataTemplateSelector), typeof(GroupEditorControl), new PropertyMetadata (default(DataTemplateSelector)));
+
+		public DataTemplateSelector ContentTemplateSelector
+		{
+			get { return (DataTemplateSelector) GetValue (ContentTemplateSelectorProperty); }
+			set { SetValue (ContentTemplateSelectorProperty, value); }
+		}
+
+		public static readonly DependencyProperty SelectorStyleProperty = DependencyProperty.Register (
+			"SelectorStyle", typeof(Style), typeof(GroupEditorControl), new PropertyMetadata (default(Style)));
+
+		public Style SelectorStyle
+		{
+			get { return (Style) GetValue (SelectorStyleProperty); }
+			set { SetValue (SelectorStyleProperty, value); }
+		}
+
+		protected override void OnSelectionChanged (SelectionChangedEventArgs e)
+		{
+			if (e.AddedItems.Count == 0)
+				return;
+
+			var presenter = (ContentPresenter) ItemContainerGenerator.ContainerFromItem (e.AddedItems[0]);
+			if (presenter == null || VisualTreeHelper.GetChildrenCount (presenter) == 0)
+				return;
+
+			var toggle = (ToggleButton) VisualTreeHelper.GetChild (presenter, 0);
```

Comment:
We are, yes. Going down to `ToggleButton` instead of a radio is about the best we can do. Any deviation from that means we've broken the control, so I'd rather it explodes.

Reply:
Right, but in later parts of the code, you're detecting that and throwing a nicer exception, so might as well do it here too...